problem_builder/public/js/mentoring_with_steps.js

```diff
@@ -0,0 +1,157 @@
+
+        nextDOM.removeAttr("disabled");
+        if (nextDOM.is(':visible')) { nextDOM.focus(); }
+
+        if (isLastChild()) {
+            tryAgainDOM.removeAttr('disabled');
+            tryAgainDOM.show();
+        }
+    }
+
+    function submit() {
+        // We do not handle submissions at this level, so just forward to "submit" method of current step
+        var child = steps[active_child];
+        child['submit'](handleResults);
+    }
+
+    function hideAllSteps() {
+        for (var i=0; i < steps.length; i++) {
+            $(steps[i].element).hide();
+        }
+    }
+
+    function cleanAll() {
+        checkmark.removeClass('checkmark-correct icon-ok fa-check');
+        checkmark.removeClass('checkmark-partially-correct icon-ok fa-check');
+        checkmark.removeClass('checkmark-incorrect icon-exclamation fa-exclamation');
+        hideAllSteps();
+    }
+
+    function displayNextChild() {
+        cleanAll();
+        findNextChild();
+        nextDOM.attr('disabled', 'disabled');
+        validateXBlock();
+    }
+
+    function findNextChild() {
+        // find the next real child block to display. HTMLBlock are always displayed
+        ++active_child;
+        var child = steps[active_child];
+        $(child.element).show();
+    }
+
+    function onChange() {
+        // We do not allow users to modify answers belonging to a step after submitting them:
+        // Once an answer has been submitted (next button is enabled),
+        // start ignoring changes to the answer.
+        if (nextDOM.attr('disabled')) {
+            validateXBlock();
+        }
+    }
+
+    function validateXBlock() {
+        var is_valid = true;
+        var child = steps[active_child];
+        if (child) {
+            is_valid = child['validate']();
+        }
+        if (!is_valid) {
+            submitDOM.attr('disabled', 'disabled');
+        } else {
+            submitDOM.removeAttr('disabled');
+        }
+        if (isLastChild()) {
+            nextDOM.hide();
+        }
+    }
+
+    function initSteps(options) {
+        for (var i=0; i < steps.length; i++) {
+            var step = steps[i];
+            step['initChildren'](options);
+        }
+    }
+
+    function handleTryAgain(result) {
+        if (result.result !== 'success')
+            return;
+
+        active_child = -1;
+        displayNextChild();
+        tryAgainDOM.hide();
+        submitDOM.show();
+        if (! isLastChild()) {
+            nextDOM.show();
+        }
+    }
+
+    function tryAgain() {
+        var handlerUrl = runtime.handlerUrl(element, 'try_again');
+        if (submitXHR) {
+            submitXHR.abort();
+        }
+        submitXHR = $.post(handlerUrl, JSON.stringify({})).success(handleTryAgain);
+    }
+
+    function initXBlockView() {
+        checkmark = $('.assessment-checkmark', element);
+
+        submitDOM = $(element).find('.submit .input-main');
+        submitDOM.bind('click', submit);
```

Comment:
@itsjeyd I believe `bind` is somewhat [deprecated](http://api.jquery.com/bind/)

    As of jQuery 1.7, the .on() method is the preferred method for attaching event handlers to a document.